include/seqan3/alignment/pairwise/policy/affine_gap_policy.hpp

```diff
@@ -56,38 +57,70 @@ class affine_gap_policy
     ~affine_gap_policy() noexcept = default;
     //!\}
 
-    /*!\brief Computes the score for current cell.
-     * \tparam        score_cell_type The type of the current cell.
-     * \tparam        cache_t         The type of the cache.
-     * \param[in,out] current_cell    The current cell in the dynamic programming matrix.
-     * \param[in,out] cache           The cache storing hot helper variables.
-     * \param[in]     score           The score of comparing the respective letters of the first and the second sequence.
+    /*!\brief Computes the score of the current cell.
+     * \tparam  matrix_entry_type The type of the current cell.
+     * \tparam  cache_type        The type of the cache.
+     * \param[in,out] current_cell The current cell in the dynamic programming matrix.
+     * \param[in,out] cache        The cache storing hot helper variables.
+     * \param[in]     score        The score of comparing the respective letters of the first and the second sequence.
      */
-    template <typename score_cell_type, typename cache_t>
-    constexpr void compute_cell(score_cell_type & current_cell,
+    template <typename matrix_entry_type, typename cache_t>
+    constexpr void compute_cell(matrix_entry_type && current_cell,
                                 cache_t & cache,
                                 score_t const score) const noexcept
     {
+        using std::get;
```

Comment:
Why do we have this so often? Souldn't we just rely on ADL?

Reply:
But std::get is not a niebloid. So we are relying on ADL but to do it in a safe way we need to additionally search in the std namespace if there is not get in the namespace of the variable.